Feat/editor core

lamarck009 opened the pull request:
## 개요

너무 복잡도가 높아서 일단 에디터 부분만 추려내서 재 업데이트 했습니다!
(7.24)

<!---- Resolves: #(Isuue Number) -->

## PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [x] 빌드 부분 혹은 패키지 매니저 수정
- [x] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)

## 💬 공유사항 to 리뷰어

일단 현재 구조는 
SplitEditorContainer
   ----  ActiveFileContent
          -----CollaborativeEditor

구조입니다.

SplitEditorContainer : 분할 패널 지원용 에디터
ActiveFileContent : 여기가 룸 프로바이더로 라이브 블록 불러오는 부분입니다.
CollaborativeEditor : 실질적인 에디터 자체 기능의 설정이 들어가는 곳입니다.

일단 다른 부분과 연동된 부분(탭이나, 상세 액션, 메서드 등)은 제외했고 최대한 줄여서 작동하는 단위를 확인하긴 했는데
기능을 좀 쳐낸거라 정상 작동하지 않으면 말씀 부탁드립니다.

filetree와 fileexplorer 부분은 파일을 열기위해선 필요해서 임시적으로 구현한거라 일단 필요하시면 사용하시되 교체하시면 될거 같습니다.
딱히 의미가 있는 코드는 없습니다.

liveblockconfig : api키와 라이브블록에서 사용되는 프로바이더등의 기능이 있는 설정파일입니다. 설정 파일에서 사용하는것을 권장해서 이렇게 수정했습니다.

split-editor-store : 기능을 최대한 쳐내고 오픈 파일만 작동하도록 한다음에 초기 상태부분만 남겨놨습니다.
이후 원본 추가 설명과 함께 pr하겠습니다.
user-store : 이건 전에 말씀드린 유저 정보용 임시 스토어입니다. 사실 지금 유저 커서도 지우고 올리는거라 의미가 없을지 모르겠지만 대강 이런식으로 연결되는 부분이라고만 파악하는 용도로 사용하시면 될거 같습니다.

editor-type : 에디터에 사용되는 타입을 정리한 파일입니다.
file-type : 파일 확장자를 위한 파일입니다.

## PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] Eslint 및 Prettier 규칙을 준수했습니다.
- [ ] origin/main에서의 최신 커밋을 확인하고 반영했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

lamarck009 committed:
feat: 코드에디터를 사용하기 위해서 임시적으로 구현한 파일탐색기
- blueprint tree를 사용해서 드래그 앤 드롭 구현
- 클릭시 파일 열기(탭 생성 가능)

lamarck009 committed:
feat: 컴포넌트를 가져와서 사용하는 코드 에디터
- 현재 가져온 컴포넌트
- TabBar : 말 그대로 탭바입니다.
- ActiveFileContent
  - IDE의 활성 파일 에디터 컨텐츠를 렌더링하는 컴포넌트
- 추가적인 변경사항
  - 콘탠츠 부분을 지웠습니다. 라이브 블록에서 내용을 가져다 쓰는 부분이라

lamarck009 committed:
feat: add code editor components
- 의존성 설치
- 라이브 블록
- 리액트 모자이크 컴포넌트(추후 스플릿 구현용)
- 리사이져 : 사이드바랑 공간 조절용

lamarck009 committed:
chore: 라이브블록 관련 기능 추가
- 사용할 아이콘도 추가했습니다.

lamarck009 committed:
refactor: 기능추가를 위해서 파일 위치 변경
- 수정한 임시 탐색기를 사이드바 폴더로 분류했습니다.
- 사이드바에 위치해있기 때문에
- 탐색기용 임시 css도 추가

lamarck009 committed:
feat: 에디터 컨텐츠 관리용
- 라이브블록 사용을 위한 키와
- 분할 모드
- 협업 에디터의 레이아웃 등을 담당합니다.

lamarck009 committed:
feat: 실제 기능이 담긴 협업 에디터와 사용자 커서
- 사용자 커서 등을 표시하는 컴포넌트를 불러옵니다.
- 모나코 에디터를 사용하는 컴포넌트를 불러옵니다.(라이브러리)

lamarck009 committed:
feat: 파일탐색기 헤더와 파일트리
- 에디터 사용을 위해서 임시 구축했습니다.

lamarck009 committed:
feat: 각종 에디터용 스토어와 탭 액션입니다
- 탭관련 : 탭바(레이아웃), 탭 아이템: 탭 모양용과 탭 관련 버튼 동작
- 탭 액션 : 탭 관련 동작 / 탭 닫기 등
- 탭드래그프로바이더 : 아직 미구현
- 에디터 스토어 관련 : 에디터 사용을 위한 임시 전역 변수들
- 유저 스토어 관련 : 유저 정보 표시용 임시 정보

lamarck009 committed:
feat: 분할 편집기의 탭 관리 기능을 구현한 Zustand 스토어
- 아직 적용 안함

lamarck009 committed:
feat: 추후 구현될 스플릿 시스템용 스토어입니다

lamarck009 committed:
feat: 분할 에디터용 유틸입니다

lamarck009 committed:
feat: 파일탭과 스플릿용 타입 정의입니다

lamarck009 committed:
feat: 탐색기를 위한 css

sjoon627 reviewed (commented):
확인했습니다! 너무 고생하셨어요 👍

혹시 다음에 pr작성해주실때는 조금 더 작은 단위로 작업해서 올려주실수 있으실까요?
3000줄은 아무래도 파악하기 힘든거 같습니다..

커밋자체도 조금 더 작은단위로 커밋하시는게 좋을거 같아요

하나의 파일에서 너무 많은 역할을 담당하는 경우가 있는데
이를 나누는 작업도 필요할거 같습니다.

interface 사용해서 타입정의된 부분은 ts하나 만들어서 관리하시는게 편하실거에요

lamarck009 committed:
chore: 라이브블록 설정파일입니다
- api키나 기능 포함

lamarck009 committed:
chore: 라이브블록 패키지 추가

lamarck009 committed:
feat: 스플릿 에디터 기능 추가

lamarck009 committed:
feat: 임시 구축한 파일탐색기입니다
- 파일 클릭을 안하면 연동을 못해서 추가했습니다.

lamarck009 committed:
feat: 에디터 코어 기능
- 에디터 코어 기능만 남긴 부분입니다.
- 일부 기능 목업화(todo)
- 특정 상태 기본값 사용

lamarck009 committed:
feat: 스토어와 타입
- 필수적인 타입과 스토어 상태만 남겼습니다.
- open 파일 액션도 있긴 해야할거 같아서 남겨뒀습니다.
- pnpm-lock.yaml은 일단 변경 사항은 있어서 올렸습니다.

lamarck009 committed:
refactor: 리팩토링 과정에서 삭제
- 전체 재구성 과정에서 삭제된 파일입니다.

lamarck009 committed:
Delete src/components/ide/editor-part/Cursors.tsx

일단 임시로 사용 안하기 때문에 삭제했습니다

lamarck009 committed:
Delete src/components/ide/editor-part/tab/TabBar.tsx

탭도 코드 복잡도가 높아서 당장에는 제외했습니다

lamarck009 committed:
Delete src/components/ide/editor-part/tab/TabItem.tsx

탭도 코드 복잡도가 높아서 당장에는 제외했습니다

lamarck009 committed:
Delete src/hooks/useFileTree.ts

복잡도가 높아서 일시 삭제했습니다.

lamarck009 commented:
최대한 쳐내고 에디터 부분 코드만 좀 축소해서 재 pr했습니다.
(일부 기능들은 일단 지워뒀습니다)
작동하는지만 확인 부탁드립니다.
